src/IRPrinter.h

```diff
@@ -50,6 +50,10 @@ struct Target;
 /** Emit a halide Target in a human readable form */
 std::ostream &operator<<(std::ostream &stream, const Target &);
 
+/** Emit a halide Func in a human readable form. */
```

Comment:
Emit a Halide Func "and its dependencies". Currently the code find all the dependent Funcs and print all of them.
Not sure if we want an option to only print one Func. The print all dependencies version was more useful for me since it prints both the boundary condition and the actual content of the gradients.

Reply:
Dumping the entire DAG seems like a weird default behavior to me. I would expect a << on a Pipeline to do that, and << on a Func to just dump that one Func

Reply:
Sounds good to me!